src/Test/StateMachine/Internal/Sequential.hs

```diff
@@ -110,6 +111,53 @@ liftGen gen pid ns = sized $ \sz -> runStateT (go sz) ns
 
     (IntRefed cmd ixref :) <$> go (pred sz)
 
+liftGen'
+  :: forall ix cmd
+  .  Ord       ix
+  => SingKind  ix
+  => DemoteRep ix ~ ix
+  => IxTraversable cmd
+  => HasResponse cmd
+  => Gen [Untyped cmd (RefPlaceholder ix)]
+  -> Pid
+  -> Map ix Int
+  -> Gen ([IntRefed cmd], Map ix Int)
+liftGen' gen pid ns = do
+  cmds <- gen
+  runStateT (go cmds) ns
+  where
+
+  go :: [Untyped cmd (RefPlaceholder ix)] -> StateT (Map ix Int) Gen [IntRefed cmd]
+  go []                   = return []
+  go (Untyped cmd0 : cmds) = do
+
+    scopes <- get
+
+    Untyped cmd <- lift . genFromMaybe $ do
```

Comment:
This will not work, if the cmd0 wants references that we don't have the this will always be return Nothing which will fail in genFromMaybe.